pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -951,6 +954,45 @@ func (m *kubeGenericRuntimeManager) killPodWithSyncResult(pod *v1.Pod, runningPo
 	return
 }
 
+func (m *kubeGenericRuntimeManager) GeneratePodStatus(event *runtimeapi.ContainerEventResponse) (*kubecontainer.PodStatus, error) {
+	podIPs := m.determinePodSandboxIPs(event.PodSandboxMetadata.Namespace, event.PodSandboxMetadata.Name, event.PodSandboxStatus)
+
+	kubeContainerStatuses := []*kubecontainer.Status{}
+	for _, status := range event.ContainersStatuses {
+		cStatus := toKubeContainerStatus(status, m.runtimeName)
+		if status.State == runtimeapi.ContainerState_CONTAINER_EXITED {
+			// Populate the termination message if needed.
+			annotatedInfo := getContainerInfoFromAnnotations(status.Annotations)
+			// If a container cannot even be started, it certainly does not have logs, so no need to fallbackToLogs.
+			fallbackToLogs := annotatedInfo.TerminationMessagePolicy == v1.TerminationMessageFallbackToLogsOnError &&
+				cStatus.ExitCode != 0 && cStatus.Reason != "ContainerCannotRun"
+			tMessage, checkLogs := getTerminationMessage(status, annotatedInfo.TerminationMessagePath, fallbackToLogs)
+			if checkLogs {
+				tMessage = m.readLastStringFromContainerLogs(status.GetLogPath())
+			}
+			// Enrich the termination message written by the application is not empty
+			if len(tMessage) != 0 {
+				if len(cStatus.Message) != 0 {
+					cStatus.Message += ": "
+				}
+				cStatus.Message += tMessage
+			}
+		}
+		kubeContainerStatuses = append(kubeContainerStatuses, cStatus)
+	}
+
+	sort.Sort(containerStatusByCreated(kubeContainerStatuses))
```

Comment:
why do we need to sort here?

Reply:
Because existing [getPodContainerStatuses](https://github.com/kubernetes/kubernetes/blob/98533290e980aaeec9d78a0efa8f9f606e3514cc/pkg/kubelet/kuberuntime/kuberuntime_container.go#L546) used by existing GetPodStatus also sorts the containers.